src/bhost_fpga.py

```diff
@@ -0,0 +1,1771 @@
+# /usr/bin/env python
+"""
+Selection of commonly-used beamformer control functions.
+
+Author: Jason Manley, Andrew Martens, Ruby van Rooyen
+"""
+"""
+Revisions:
+2012-10-02 JRM Initial
+2013-02-10 AM basic boresight
+2013-02-11 AM basic SPEAD
+2013-02-21 AM flexible bandwidth
+2013-03-01 AM calibration
+2013-03-02 RR fbfExceptions
+2013-03-05 AM SPEAD metadata
+\n"""
+
+import numpy
+import inspect
+import logging
+import struct
+import socket
+import spead64_48 as spead
+from casperfpga import utils
+
+LOGGER = logging.getLogger(__name__)
+
+class fbfException(Exception):
+    def __init__(self, errno, msg, trace=None, logger=None):
+        self.args = (errno, msg)
+        self.errno = errno
+        self.errmsg = msg
+        self.__trace__ = trace
+        if logger:
+            logger.error('BFError: %s\n%s' % (msg, trace))
+
+class FrequencyBeamformer:
```

Comment:
Better to class FrequencyBeamformer(object): ... otherwise you get 'old-style' python classes that sometimes behave differently to what you might expect.